compiler/rustc_const_eval/src/interpret/intrinsics.rs

```diff
@@ -69,6 +69,9 @@ pub(crate) fn eval_nullary_intrinsic<'tcx>(
             ty::Alias(..) | ty::Param(_) | ty::Placeholder(_) | ty::Infer(_) => {
                 throw_inval!(TooGeneric)
             }
+            ty::Pat(..) => {
+                unimplemented!("pattern types need to calculate pattern from their pattern")
```

Comment:
```suggestion
                unimplemented!("pattern types need to calculate available variants from their pattern")
```

Reply:
this suggestion has not been applied afaict